source/module_esolver/esolver_ks_lcao.cpp

```diff
@@ -689,9 +689,14 @@ void ESolver_KS_LCAO<TK, TR>::iter_init(const int istep, const int iter)
         this->p_hamilt->refresh();
     }
 
+    static bool drho_condition = false;
     // run the inner lambda loop to contrain atomic moments with the DeltaSpin method
-    if (GlobalV::sc_mag_switch && iter > GlobalV::sc_scf_nmin)
+    if (GlobalV::sc_mag_switch && iter > GlobalV::sc_scf_nmin && (this->drho < GlobalV::sc_scf_start || drho_condition))
```

Comment:
GlobalV will be removed in near future.

Reply:
@mohanchen Thanks for your comments. I will make a new PR to achieve this. So close this one now.